mara_db/shell.py

```diff
@@ -79,6 +78,16 @@ def __(db: dbs.RedshiftDB, timezone: str = None, echo_queries: bool = None):
             + (db.database or ''))
 
 
+@query_command.register(dbs.BigQueryDB)
+def __(db: dbs.BigQueryDB, timezone: str = None, echo_queries: bool = None):
+    echo_queries = None
+    assert all(v is None for v in [timezone, echo_queries]), f"unimplemented parameter for BigQueryDB"
+
+    return ('bq query'
+            + (f' --use_legacy_sql=' + ('true' if db.use_legacy_sql else 'false'))
+            + ' --quiet --headless -n=1000000000 ')
```

Comment:
Is there a need for quiet, headless or the 'n' argument?

in addition, I think we should pass here over the project id, default location and dataset id, see also https://github.com/hz-lschick/mara-db-bigquery/blob/master/mara_db_bigquery/shell.py

Reply:
Yes, this can be improved. 
**--headless** Relatively minor because it doesn't seem to affect but I would keep for disabling any kind of user-interaction mode based on the description in the official docs (https://cloud.google.com/bigquery/docs/reference/bq-cli-reference)
**--quiet** similar concept. It doesn't seem to affect on some simple testing that I tried but I would keep since we are manipulating data through stdin/out and we don't want any metadata there.
About the **n** argument, I couldn't find another way to disable the default limits and I have this ugly thing. We need to explore more there and do it in a proper way. 
How do you actually do that? Please share if you more insights on this!

Reply:
Headless and Quite sounds good to me. About **n**: The question is what do you want to do?; e.g. what command is running into the limit? Did you look at `bq load` or `bq extract`?

When this is really necessary, I would consider rather building an extension class to mara_pipelines.commands.sql.ExecuteSQL, e.g. 'ExecuteBigQuerySql' which has an additional parameter which is then passed over to 'n' . But this would make sence rather in a separate repository.

Reply:
From the cli docs on bq-query command:
```
--max_rows or -n
    An integer specifying the number of rows to return in the query results. The default value is 100
```
The problem is that the basic **bq query** command is under the default 100 limit and I can not find a way to have unlimited results there...

Reply:
I currently just use query_command for executing DDL executions, so, I don't have this problem. When you want to execute huge chunks of data, I recommend using [`bq extract`](https://cloud.google.com/bigquery/docs/exporting-data) which has other limits.

I am fine with this parameter in general but think it should be somehow optional. Each query has its cost in BigQuery, so, I think when you go over a limit, this should be actively 'activated' and not just be in the mara BigQuery module implementation to avoid extensive costs by accident.

You can configure local default parameters for the `bq` cli with an config file which you place locally on your server. This could be the easy solution for you. See here: https://cloud.google.com/bigquery/docs/bq-command-line-tool#setting_default_values_for_command-line_flags

Reply:
**bq extract** is a good idea but its limited only to work with the cloud storage and there might be use-cases that need the data locally. We should consider this option though.

There is no need to have such a user option/argument in the command itself. The limit can be specified in the query itself so the user has total control on that.
Although, if we want to have global settings for some BQ parameters (which I agree makes sense due to costs by accident etc), I like the solution with the default config file option where we can define/hide some of our defaults (starting from this ugly n value) :)